### ISSUE TITLE
Save requesting image to local and return a resultant image with bounding box?

### ISSUE THREAD

**Issue by jiapei-nexera** [pytorch/serve#1710]
### 🚀 The feature

By requesting
`curl http://192.168.1.100:8080/predictions/alexnet -T ./serve/examples/image_classifier/kitten.jpg`,

Wondering if there is an option to:

- save the uploaded **kitten.jpg** to a local folder on http://192.168.1.100 ??
- return an image with bounding box to the requesting web browser ?

### Motivation, pitch

As described in **The feature**, the motivation is quite obvious.

### Alternatives

N/A

### Additional context

N/A

**Comment by msaroufim** [comment]
The way you'd support this is extending this function in your own custom handler. So inherit `ImageClassifier` and then add support in your own `postprocess()` function support for writing an image to disk or returning whatever you like including a bounding box https://github.com/pytorch/serve/blob/master/ts/torch_handler/image_classifier.py#L35

If you need more pointers lmk

**Comment by TillmannRheude** [comment]
I tried following your explanation @msaroufim and wrote my own class by inheriting ImageSegmenter. By adding something like `torch.save(tensorfile, filename.pt)`, into `postprocess`, I cannot identify where the tensor is saved. Further, by changing the return of `postprocess `in order to be able to do `curl -o`, an error message in the model_log is created which states `Invalid return type: <class 'torch.Tensor'>.`

Could you state a minimal example how to modify the postprocess function in order to save an image/segmentation?

**Comment by agunapal** [comment]
@TillmannRheude  Will get back to you with an example

**Comment by agunapal** [comment]
@TillmannRheude  The following code worked for me with the AlexNet example.
I modified the inference code to return a tuple of the inference result and input image
We can unpack the data in the postprocess method and do the necessary post processing operation on the image and save it

`   def inference(self, data, *args, **kwargs):
        marshalled_data = data.to(self.device)
        with torch.no_grad():
            results = self.model(marshalled_data, *args, **kwargs)
        return (results, data)

    def postprocess(self, data):
        data, inp_img = data[0], data[1][0]
        
        ps = F.softmax(data, dim=1)
        probs, classes = torch.topk(ps, self.topk, dim=1)
        probs = probs.tolist()
        classes = classes.tolist()

        # post process the image
        op_image = inp_img

        # save image to file
        save_image(op_image, "/home/ubuntu/op_img1.png")
        
        return map_class_to_label(probs, self.mapping, classes)`

**Comment by TillmannRheude** [comment]
Thank ya @agunapal, it worked like a charm - My fault was not to define "/home/.." in front of the path for the image.

**Comment by agunapal** [comment]
Thanks for confirming. Closing the issue
